cssparse/shared/src/main/scala/cssparse/CssParser.scala

```diff
@@ -0,0 +1,205 @@
+package cssparse
+
+import fastparse.all._
+
+import scala.collection.mutable.ArrayBuffer
+
+// According to https://www.w3.org/TR/css-syntax-3/
+
+object CssTokensParser {
+
+  val comment = P( "/*" ~/ (!"*/" ~ AnyChar).rep ~/ "*/")
+
+  val newline = P( "\n" | "\r\n" | "\r" | "\f")
+
+  val whitespace = P( " " | "\t" | newline)
+
+  val hex_digit = P( CharIn('0' to '9', 'a' to 'f', 'A' to 'F') )
+
+  val escape = P( "\\" ~ ((!(newline | hex_digit) ~ AnyChar) |
+    (hex_digit.rep(min=1, max=6) ~ whitespace.?)) )
+
+  val whitespace_token = P( comment | whitespace.rep(1) )
+
+  val ws = P( whitespace_token.rep )
+
+  val ident_token = P( ("-".? ~ (CharIn('a' to 'z', 'A' to 'Z', "_") | escape) ~
```

Comment:
Break long lines like this into multiple sub-parsers, combined in one final parser, to keep each parser to one line if possible. The only multi-line parsers should have be relatively simple (e.g. A bunch of named parsers separated by `|`s) and not a complicated mix of `|`s and `~`s. Scope the sub-parsers within this `val` to keep them local